spec/services/paginated_comments_spec.rb

```diff
@@ -10,21 +10,24 @@
   let(:sorted_ids)     { Comment.order(id: :desc).pluck(:id) }
   let(:params) { {limit: limit, position: start_at} }
 
-  subject { described_class.new(debate: debate, direction: :backward, params: params).retrieve_comments }
+  subject { described_class.new(comments_relation: debate.retrieve_comments,
+                                direction: :backward,
+                                params: params).to_h }
 
   context 'When comments are created' do
     before { create_list(:comment, comment_count, debate: debate, status: :accepted) }
 
-    it 'returns list of comments' do
-      expect(subject.count).to eq(limit)
+    it 'returns list of comments with limit' do
```

Comment:
I would split this to two separate tests, one checking limit and other checking next position. Also i would check here if position is numeric only